src/material/timepicker/timepicker.md

```diff
@@ -1 +1,310 @@
+
+### Timepicker forms integration
+
+The timepicker input integrates with the `@angular/forms` module by providing itself as a
+`ControlValueAccessor` and a `Validator` (see [Input validation](#input-validation) below for more
+information). When the user types in a new time or selects one from the dropdown, the
+time will be set on the date object which is the current value of the form control. If the form
+control doesn't have a value, the timepicker will create one with today's date and the selected
+time.
+
+<!-- example(timepicker-forms) -->
+
+
+### Integration with `MatDatepicker`
+
+Material's datepicker and timepicker components can operate over the same value, allowing for a
+combined datetime picker to be implemented. When binding the two components to the same value, the
+datepicker will set the entire date object while the timepicker will only modify the time portion
+of it.
+
+<!-- example(timepicker-datepicker-integration) -->
+
+### Input validation
+
+The timepicker input checks that the value typed in by the user is a valid time string and
+whether it fits into the specified bounds.
+
+If the user types in an invalid time string (for example `abc` or `24:67`), the timepicker input
+will report the `matTimepickerParse` error. The string is parsed using the `parseTime` method of
+the [the current date implementation](#choosing-a-date-implementation-and-format-settings).
+
+The timepicker input also checks that the value typed in by the user fits within the minimum and
+maximum bounds set through the `matTimepickerMin` and `matTimepickerMax` inputs. They accept either
+a date object with a specific time or a time string. The inputs also control which times will be
+shown inside of the dropdown menu. For example, setting `matTimepickerMin="12:30"` and
+`matTimepickerMax="21:25"` will allow the user to only select a time between 12:30 in the afternoon
+and 9:25 in the evening. If the value is outside of those bounds, either a `maxTimepickerMin` or
+`matTimepickerMax` error will be reported to the value accessor.
+
+<!-- example(timepicker-validation) -->
+
+
+### Customizing the dropdown options
+
+By default the `mat-timepicker` dropdown shows options at 30 minute intervals. You can customize the
+list of options either by setting an interval or providing a custom set of options.
+
+The easiest way is to change the options is to pass the `interval` input to `mat-timepicker` with
+an interval string which will be used when generating the options. For example,
+`<mat-timepicker interval="90m"/>` will show the options at 90 minute intervals, starting from the
+minimum time and ending at the maximum. Valid interval strings include:
+* A number which will be interpreted as minutes, e.g. `interval="50"` represents 50 minutes.
+* A number with short units, for example `30m` represents 30 minutes while `5h` is 5 hours.
+Supported short units include `h`/`H` for hours, `m`/`M` for minutes and `s`/`S` for seconds.
```

Comment:
It might be more clear if its `h` or `H` instead of `h`/`H` for the different short units.